pkg/allocator/cache.go

```diff
@@ -200,12 +200,31 @@ func (c *cache) onDeleteLocked(id idpool.ID, key AllocatorKey, recreateMissingLo
 
 					ctx, cancel := context.WithTimeout(ctx, backendOpTimeout)
 					defer cancel()
+					ifLocalKeyStillExistsLocked := func(fn func() error) error {
+						// We ensure we hold the slaveKeysMutex while checking if the localKey is still in use, and while we are
+						// doing the create/update operation. We do this to avoid re-creating a newly deleted slave key, that can
+						// potentially result in leaking it - as the identity gc in the operator won't delete slave keys.
+						a.slaveKeysMutex.Lock()
```

Comment:
Not sure if we really want to do this here to reduce the likelyhood of deadlocks, or if we just do `a.localKeys.lookupID(id); key == nil || key.GetKey() != value.GetKey()` and nothing more